### PR TITLE
Add authorization

### PR DESCRIPTION
**Pull request by SelmaNdapanda**
### In this pull request, @yonas44 and I implemented the following:  :arrow_down: 

- Install CanCanCan in your project.  :heavy_check_mark: 
- Add a `role` column to the users table. Remember to use a migration for this.  :heavy_check_mark: 
- A user can delete a post if it is theirs or if they have an admin role (column `role` has value `"admin"`). Use CanCanCan for this authorization.  :heavy_check_mark: 
    - For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column `role` has value `"admin"`). Use CanCanCan for this authorization.  :heavy_check_mark: 
    - For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/views/posts/show.html.erb [hunk 1]
```diff
@@ -13,6 +13,9 @@
     <ul class="container">
       <% @post.comments.each do |comment| %>
         <li><%= comment.author.name %>: <%= comment.text %></li>
+        <% if can? :destroy, Comment %>
+          <%=button_to "delete", user_post_comment_path(current_user, @post, comment), method: :delete %>
+        <% end %>
```
Comment (elisha2kyakpo1) [inline app/views/posts/show.html.erb]: - [ ] I created two accounts in two browsers, and I was able to delete another person's comment, kindly fix this issue. I think it is not even good to show a delete button to users who are not eligible to delete a comment.

roles

post-delete

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -12,6 +12,14 @@
         <span><em><b>Comments: <%= post.comments_counter%></b></em></span>
         <span><em><b>Likes: <%= post.likes_counter%></b></em></span>
       </div>
+      <ul>
+        <% if can? :destroy, Post %>
+          <li><%= button_to "del", user_post_path(post.author, post.id), method: :delete  %></li>
+        <%end%>
```
Comment (elisha2kyakpo1) [inline app/views/posts/index.html.erb]: - [ ] Let's fix the relationships for a post table, comment table, and like table. We should instruct a parent table (post table) that when a post with a like(s) or with a comment(s) is deleted, even its likes or comments should be deleted. We can do this at the model level or in our migration files. I got the error in the image below.

like

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @SelmaNdapanda and @yonas44, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description look good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb (hunk 2 above)
Comment (yonas44) [reply app/views/posts/index.html.erb]: Okay thank you for the feedback @elisha2kyakpo1, we will address them all.

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi team, 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- GitHub flow is used correctly. ✔
- Good PR summary. ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**